test/integration/eds_integration_test.cc

```diff
@@ -164,5 +164,29 @@ TEST_P(EdsIntegrationTest, BatchMemberUpdateCb) {
   EXPECT_EQ(1, member_update_count);
 }
 
+TEST_P(EdsIntegrationTest, StatsReadyFilter) {
```

Comment:
> @jmarantz - any idea why the new eds_ready_filter can't find the stat in the root scope? Some very hacky debug logging suggests that the "cluster.cluster_0.membership_healthy" stat is added to scopes whose parent is the root scope, but querying the root scope for that stat returns nothing.

I'm not sure my knowledge of the stats system is still surprisingly narrow and I learn new things all the time. But I'd guess that it's not supposed to be in the root scope and there's a scope created  for "cluster" or perhaps "cluster.cluster_0".

But doesn't your new function, which I was assuming you'd test directly below, loop over all the scopes?

Reply:
Yes, the getGauge/Counter/Histogram implementations loop over all the scopes.

Is the assumption that we'd truncate the StatName, e.g. look for "cluster_0.membership_healthy" in the "cluster" scope instead of the full "cluster.cluster_0.membership_healthy"?

Reply:
Yeah the names inside the scopes are not relative to the scope, like (say) a unix file in a directory.

The objects in scope "foo." have names with the "foo." in them.

Reply:
All right, I figured out part of my problem looking up stats: I was adding the prefix to an already "fully-qualified" stat. Is there a way to ask if a StatName starts with the prefix and join the prefix if not?

This also made me realize that the result is ambiguous if someone calls findCounter for "foo.bar.counter" where there are overlapping scopes (e.g. counter "bar.counter" in scope "foo" and counter "counter" in scope "foo.bar"). Not sure how much we care, or what a good way to communicate that is.

Reply:
I think there's not an ambiguity; the names we do lookups on are all fully qualified. They may exist in more than one scope, but they'll all point to the same underlying stat, so it doesn't matter which scope it's found in.

Reply:
Ah, got it. Thanks for the clarification!